internal/daemon/controller/interceptor.go

```diff
@@ -355,7 +355,14 @@ func aliasResolutionInterceptor(
 		}
 		interceptorCtx, err = alias.ResolveRequestIds(interceptorCtx, reqMsg, r)
 		if err != nil {
-			return nil, err
+			// Since this is intercepted prior to checking that the requester
+			// is authorized to make the request, returning a 404 here and a 403
+			// when the request is checked in the handler would expose which
+			// aliases exist. Instead, we return a forbidden error here so there
+			// is no way to distinguish between a missing alias and an existing
+			// alias that points to a destination the requester is not allowed
+			// to perform an action on.
+			return nil, handlers.UnauthenticatedError()
```

Comment:
May be worth adding a little note here to say that `handlers.UnauthenticatedError()` turns in to HTTP Status 401 _Unauthorized_ (thanks HTTP), since the inconsistency of using `Unauthenticated` and `Unauthorized` is easy to be confused by.